pkg/telemetry/events/events.go

```diff
@@ -1,19 +1,106 @@
 package events
 
 import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/KubeHound/pkg/telemetry/log"
 	kstatsd "github.com/DataDog/KubeHound/pkg/telemetry/statsd"
+	"github.com/DataDog/KubeHound/pkg/telemetry/tag"
 	"github.com/DataDog/datadog-go/v5/statsd"
 )
 
 const (
-	DumperRun  = "kubehound.dumper.run"
-	DumperStop = "kubehound.dumper.stop"
+	EventActionFail   = "fail"
```

Comment:
I believe it would be a nice improvement to use these as a custom `EventAction` type instead of string.

This way, you can enforce the strings to match your expectation.
The defined api contract is a bit "weird" here imo, as it contains package specific code (`PushEventIngestSkip` is tied to the `Ingestor`, directly in its name). I'd not mind if it was is a struct / map "as data" for collocation purpose, but as an exposed API it's couple the 2 packages.

You could refactor all of these individual function into a single one like so (following code not tested) to avoid the duplication for each event and makes it simpler and easier to udpate overall:

```go
type EventAction string

const (
	EventActionFail   = "fail"
	EventActionInit   = "init"
	EventActionStart  = "start"
	EventActionSkip   = "skip"
	EventActionFinish = "finish"
)

func (ea EventAction) Message() string{
	return map2msg[ea]
}

func (ea EventAction) Tags() []string {
	tags := tag.GetDefaultTags(ctx)
	tags = append(tags, fmt.Sprintf("%s:%s", ActionTypeTag, ea))
	return tags 
}
func (ea EventAction) Level() statsd.Level {
	return statsd.Info
}

// Could also be a format stirng template in this case, if needed?
var map2msg = map[EventAction]string{
	EventActionFail = "Ingestor/grpc endpoint init failed",
	EventActionInit = "" //...
	// ...
}

func PushEvent(ctx context.Context, action EventAction) error{
	return kstatsd.Event(&statsd.Event{
		Title:     action.Title(),
		Text:      action.Message(),
		Tags:      action.Tags(),
		AlertType: action.Level(),
	})
}

Reply:
ok make sense but some events requires args that needs to be passed. So they will need a custom PushEvent wrapper for those specific.

Reply:
I'd think a `Option` (variadic) type would be enough and allow basic definition to not need it? But I haven't tried